src/frontend/screens/Settings/CreateTestData.tsx

```diff
@@ -0,0 +1,305 @@
+import {Preset} from '@mapeo/schema';
+import {useMutation, useQueryClient} from '@tanstack/react-query';
+import {lengthToDegrees} from '@turf/helpers';
+import {randomPosition} from '@turf/random';
+import {LocationObject} from 'expo-location';
+import {type BBox} from 'geojson';
+import React, {forwardRef} from 'react';
+import {Controller, useForm} from 'react-hook-form';
+import {StyleSheet, TextInput, ToastAndroid, View} from 'react-native';
+import {UIActivityIndicator} from 'react-native-indicators';
+
+import {useActiveProject} from '../../contexts/ActiveProjectContext';
+import {useDeviceInfo} from '../../hooks/server/deviceInfo';
+import {OBSERVATION_KEY} from '../../hooks/server/observations';
+import {usePresetsQuery} from '../../hooks/server/presets';
+import {useLocation} from '../../hooks/useLocation';
+import {LIGHT_GREY, RED, WHITE} from '../../lib/styles';
+import {Button} from '../../sharedComponents/Button';
+import {LocationView} from '../../sharedComponents/Editor/LocationView';
+import {ScreenContentWithDock} from '../../sharedComponents/ScreenContentWithDock';
+import {Text} from '../../sharedComponents/Text';
+
+const DISTANCE_BUFFER_KM = 50;
+
+const BASE_NUMBER_INPUT_RULES = {
+  min: 1,
+};
+
+export function CreateTestDataScreen() {
+  const presetsQuery = usePresetsQuery();
+  const locationState = useLocation({maxDistanceInterval: 0});
```

Comment:
what should be done in the case that it returns `null`? unlike `useLocation()`, we'll be stuck with that value since it's not subscription-based. in that case we'd potentially have to allow generating observations that don't have a location, which is definitely valid but will require some more work around how the implementation current works.

Reply:
maybe i should already support observations without location anyways? just didn't want add more complex UI than necessary if that wasn't going to be useful initially. maybe can be done as a follow-up (in addition to the potentially using `useLastKnownLocation()` instead)

Reply:
do we not have this problem with useLocation() as well? Can `useLocation()` also be null? Its not the biggest deal, i think if there is too much work in changing it, happy to just leave it as is

Reply:
yeah we could, hence the note about potentially supporting creating observations without a location. at least in that case, there's a chance that the location will become a valid value though since it's subscription-based.

> Its not the biggest deal, i think if there is too much work in changing it, happy to just leave it as is

yeah i think for now it's fine to keep as is but could definitely be a follow-up!